zookeeper-server/src/main/java/org/apache/zookeeper/common/X509Util.java

```diff
@@ -167,47 +222,50 @@ public SSLContext createSSLContext(ZKConfig config) throws SSLContextException {
         KeyManager[] keyManagers = null;
         TrustManager[] trustManagers = null;
 
-        String keyStoreLocationProp = config.getProperty(sslKeystoreLocationProperty);
-        String keyStorePasswordProp = config.getProperty(sslKeystorePasswdProperty);
+        String keyStoreLocationProp = config.getProperty(sslKeystoreLocationProperty, "");
```

Comment:
Since empty string is not a valid keystore type, I treat null value and empty string values equivalently. Doing it this way simplifies the check below (only need to check for empty string instead of checking both null and empty).

Reply:
Sorry, I misread this comment. You are talking about the location, not the store type. But it's the same thing - null location and empty location can be treated the same way, so by defaulting to empty string here we don't need to check for 2 different options below. The previous code actually didn't check for empty location at all.